src/common/com/intellij/plugins/haxe/util/HaxeAbstractEnumUtil.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2000-2013 JetBrains s.r.o.
+ * Copyright 2014-2016 AS3Boyan
+ * Copyright 2014-2014 Elias Ku
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.util;
+
+import com.intellij.plugins.haxe.lang.psi.*;
+import com.intellij.plugins.haxe.model.type.HaxeClassReference;
+import com.intellij.plugins.haxe.model.type.ResultHolder;
+import com.intellij.plugins.haxe.model.type.SpecificHaxeClassReference;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+public class HaxeAbstractEnumUtil {
+
+  @Contract("null -> false")
+  public static boolean isAbstractEnum(@Nullable PsiClass clazz) {
+    if(clazz != null && clazz instanceof HaxeAbstractClassDeclaration) {
+      final HaxeMacroClassList metaList = ((HaxeAbstractClassDeclaration)clazz).getMacroClassList();
+      final List<HaxeMacroClass> metas = metaList != null ? metaList.getMacroClassList() : null;
+      if(metas != null) {
+        for (HaxeMacroClass meta : metas) {
+          if (meta.getText().equals("@:enum")) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * If this element suitable for processing by `@:enum abstract` logic
+   * IMPORTANT: This method doesn't check if this field inside `@:enum abstract`!
+   */
+  @Contract("null -> false")
+  public static boolean checkField(@Nullable PsiElement element) {
+    if(element != null && element instanceof HaxeVarDeclarationPart) {
+      final HaxeVarDeclarationPart decl = (HaxeVarDeclarationPart)element;
+       if(decl.getPropertyDeclaration() == null && !decl.isStatic()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Nullable
+  public static HaxeClassResolveResult resolveFieldType(@Nullable PsiElement element) {
```

Comment:
I've added comment for Util class - `Extensions for resolving and analyzing Haxe @:enum abstract type`, because it has a bunch of utilities to
1. general resolving
2. semantic analysis